misc/scripts/install_linuxbsd_deps.sh

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env sh
+set -eu
+IFS=$(printf '\n\t')
+
+print_info() {
+	printf "\e[1m%b\e[0m\n" "$1"
+}
+
+print_success() {
+	printf "\e[1;92m%b\e[0m\n" "$1"
+}
+
+print_error() {
+	printf "\e[1;91mERROR: \e[22m%b\e[0m\n" "$1"
+}
+
+# The detection below also covers derivatives such as Linux Mint and Manjaro.
+if [ -f "/etc/alpine-release" ]; then
+	distribution="alpine"
+	distribution_name="\e[94mAlpine"
+elif [ -f "/etc/arch-release" ]; then
+	distribution="arch"
+	distribution_name="\e[96mArch Linux"
+elif [ -f "/etc/fedora-release" ]; then
+	distribution="fedora"
+	distribution_name="\e[94mFedora"
+elif [ -f "/etc/gentoo-release" ]; then
+	distribution="gentoo"
+	distribution_name="\e[95mGentoo"
+elif [ -f "/etc/mageia-release" ]; then
+	distribution="mageia"
+	distribution_name="\e[94mMageia"
+elif grep -q openSUSE /etc/os-release; then
+	distribution="opensuse"
+	distribution_name="\e[92mopenSUSE"
+elif [ -f "/etc/solus-release" ]; then
+	distribution="solus"
+	distribution_name="\e[96mSolus"
+elif [ -f "/etc/lsb-release" ]; then
+	# Check for lsb-release last as distributions like Mageia also have this file.
+	# Ubuntu must be checked before checking for Debian as Ubuntu also has `/etc/debian_version`.
+	distribution="ubuntu"
+	distribution_name="\e[38;5;208mUbuntu"
+elif [ -f "/etc/debian_version" ]; then
+	distribution="debian"
+	distribution_name="\e[91mDebian"
+elif [ "$(uname -s)" = "FreeBSD" ]; then
+	distribution="freebsd"
+	distribution_name="\e[91mFreeBSD"
+elif [ "$(uname -s)" = "OpenBSD" ]; then
+	distribution="openbsd"
+	distribution_name="\e[93mOpenBSD"
+elif [ "$(uname -s)" = "NetBSD" ]; then
+	distribution="netbsd"
+	distribution_name="\e[38;5;208mNetBSD"
+else
+	print_error "Couldn't detect your Linux/*BSD distribution. No dependencies have been installed.\n       You'll have to install dependencies manually using your distribution's package manager:\n\n       https://docs.godotengine.org/en/latest/contributing/development/compiling/compiling_for_linuxbsd.html"
+	exit 1
+fi
+
+print_info "Detected Linux/*BSD distribution: $distribution_name"
+
+if [ -n "$(command -v sudo)" ]; then
+	# Use `sudo` when needed.
+	sudo="sudo"
+	print_info "Installing dependencies with sudo... (you may be prompted for your administrator password)"
```

Comment:
A small remark about the message printed: `sudo` asks for the user's password, not the administrator's one :slightly_smiling_face: